### DISCUSSION TITLE
TCP Provider: Error code 0x274C while running HammerDB 3.3 TPC-H test.

### DISCUSSION THREAD

**Discussion by zurca** [TPC-Council/HammerDB#208] [Q&A] [answered]
Do we know what could cause this error mid execution?

Vuser 5 successfully completed 39 queries before hitting this error. All other Vusers successfully completed the test. Below is a portion of the output, including some successful queries by Vuser 6 after Vuser 5 hit the error.

What I am looking for is a way to prevent or minimize this error from happening. Thanks!

```
^M^[[10C^MVuser 5:query 5 completed in 31.822 seconds
^M^[[10C^MVuser 5:Executing Query 8 (17 of 22)
^M^[[10C^MVuser 7:query 13 completed in 35.93 seconds
^M^[[10C^MVuser 7:Executing Query 5 (15 of 22)
^M^[[10C^MVuser 2:query 20 completed in 24.657 seconds
^M^[[10C^MVuser 2:Executing Query 3 (20 of 22)
^M^[[10C^MVuser 5:[Microsoft][ODBC Driver 17 for SQL Server]TCP Provider: Error code 0x274C [Microsoft][ODBC Driver 17 for SQL Server]Communication link failure (executing the statement)
^M^[[10C^MVuser 5:query 8 completed in 40.053 seconds
^M^[[10C^MVuser 5:Executing Query 20 (18 of 22)
^M^[[10C^MVuser 5:[Microsoft][ODBC Driver 17 for SQL Server]The connection is broken and recovery is not possible. The connection is marked by the client driver as unrecoverable. No attempt was made to restore the connection. (preparing statement)
^M^[[10C^MVuser 5:query 20 completed in 1.001 seconds
^M^[[10C^MVuser 5:Executing Query 12 (19 of 22)
^M^[[10C^MVuser 5:[Microsoft][ODBC Driver 17 for SQL Server]The connection is broken and recovery is not possible. The connection is marked by the client driver as unrecoverable. No attempt was made to restore the connection. (preparing statement)
^M^[[10C^MVuser 5:query 12 completed in 1.002 seconds
^M^[[10C^MVuser 5:Executing Query 17 (20 of 22)
^M^[[10C^MVuser 5:[Microsoft][ODBC Driver 17 for SQL Server]The connection is broken and recovery is not possible. The connection is marked by the client driver as unrecoverable. No attempt was made to restore the connection. (preparing statement)
^M^[[10C^MVuser 5:query 17 completed in 1.001 seconds
^M^[[10C^MVuser 5:Executing Query 10 (21 of 22)
^M^[[10C^MVuser 6:query 19 completed in 60.151 seconds
^M^[[10C^MVuser 6:Executing Query 20 (20 of 22)
^M^[[10C^MVuser 5:[Microsoft][ODBC Driver 17 for SQL Server]The connection is broken and recovery is not possible. The connection is marked by the client driver as unrecoverable. No attempt was made to restore the connection. (preparing statement)
```

**Comment by sm-shaw** [comment, marked as answer]
Error 0x274C/10060 "Communication link failure" is a network error where communication has been lost. Note that Vuser 6 continuing should not be an indication for Vuser 5. Each Virtual User loads the ODBC library and makes a connection separately so in this case only Vuser 5 lost connection. This depends on your network configuration eg whether you are using TCP, Azure or whether for example a connection timedout etc.

**Reply by zurca** [reply]
Thank you for the thorough explanation sm-shaw! I suspected this was likely a network issue. Do you think using TCP would help avoiding hitting this problem? This is not in Azure.

**Comment by sm-shaw** [comment]
It is possible TCP could help. This error is occurring at the level of the ODBC driver below HammerDB in this case ODBC Driver 17 for SQL Server. Therefore, another troubleshooting potential is to open the ODBC Data Source Administrator and enable tracing to see if that gives more information when the error occurs.
